src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -495,6 +496,52 @@ async def sharing_task_determine_locked(self, data, locked_datasets):
         else:
             return await super().sharing_task_determine_locked(data, locked_datasets)
 
+    @private
+    async def get_options(self, options):
+        # Do not use `SharingService.get_options`, we'll extend rows ourselves
+        return await CRUDService.get_options(self, options)
+
+    @filterable
+    async def query(self, filters, options):
+        result = await super().query(filters, options)
+
+        get = False
+        if not isinstance(result, list):
+            get = True
+            result = [result]
+
+        if result:
+            try:
+                about_to_lock_dataset = await self.middleware.call('cache.get', 'about_to_lock_dataset')
+            except KeyError:
+                about_to_lock_dataset = None
+
+            if len(result) == 1 and result[0]['type'] == 'DISK' and about_to_lock_dataset is None:
+                # Special optimized case
+                try:
+                    dataset = await self.middleware.call(
+                        'zfs.dataset.query',
+                        [['id', '=', result[0]['disk'][5:]]],
+                        {
+                            'extra': {'properties': ['encryption', 'keystatus']},
```

Comment:
Specifying `properties` has no effect in 12.0 because for single `id` case this is not respected as we are not using `datasets_serialized` endpoint.